### PR TITLE
Advanced Building Blocks - Enumerables

### PR DESCRIPTION
**Pull request by luqmanmusah**
In this project, we were required to implement our own version of Ruby’s enumerable methods. Full task description:

#Create a script file to house your methods and run it in IRB to test them later.
#Add your new methods onto the existing Enumerable module. Ruby makes this easy for you because any class or module can be added to without trouble ... just do something like:
#Create #my_each, a method that is identical to #each but (obviously) does not use #each. You'll need to remember the yield statement. Make sure it returns the same thing as #each as well.
#Create #my_each_with_index in the same way.
#Create #my_select in the same way, though you may use #my_each in your definition (but not #each).
#Create #my_all? (continue as above)
#Create #my_any?
#Create #my_none?
#Create #my_count
#Create #my_map
#Create #my_inject
#Test your #my_inject by creating a method called #multiply_els which multiplies all the elements of the array together by using #my_inject, e.g. multiply_els([2,4,5]) #=> 40
#Modify your #my_map method to take a proc instead.
#Modify your #my_map method to take either a proc or a block. It won't be necessary to apply both a proc and a block in the same #my_map call since you could get the same effect by chaining together one #my_map call with the block and one with the proc. This approach is also clearer, since the user doesn't have to remember whether the proc or block will be run first. So if both a proc and a block are given, only execute the proc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,119 @@
-# Advanced_Building_Blocks_enumerables
\ No newline at end of file
+# Building_Blocks# bubble_sort
+
+In this project I re-made some of the most used Ruby Enumerable classes. The project is done as part of microverse main curriculum from the Ruby Section.
```
Comment (arslanbisharat) [inline README.md]: - [ ] [Optional] Instead of re-made you can say re-created.

File: README.md [hunk 2]
```diff
@@ -1 +1,119 @@
...
+
+1. my_each for each
+2. my_select for select
+3. my_all? for all?
+4. my_any? for any?
+5. my_none? for none?
+6. my_count for count
+7. my_count() for count()
+8. my_map for map
+9. my_inject for inject
```
Comment (arslanbisharat) [inline README.md:17]: - [ ] [Optiona] You can add highlight here what are you talking about e.g, Methods

File: README.md [hunk 3]
```diff
@@ -1 +1,119 @@
...
+Ruby installed on your local machine.
+Terminal or similar to execute the program.
+
+## Testing
+In this project the testing was done using RSpec to test all game logic methods.
+## Installation
+
+1. Install Ruby on your machine
+
+2. Download and Unzip the folder containing this project, or clone it.
```
Comment (arslanbisharat) [inline README.md]: - [ ] [Optional] For this project we cannot download it to run the test. You can change it to clone only.

File: README.md [hunk 4]
```diff
@@ -1 +1,119 @@
...
+
+3. See "Usage" below to see how to use the classes
+
+4. In the terminal, navigate to the folder containing the file and run "ruby index.rb"
```
Comment (arslanbisharat) [inline README.md]: - [ ] [Optional] I would recommend change the name of your` index.rb` to `enumerables.rb`

File: README.md [hunk 5]
```diff
@@ -1 +1,119 @@
...
+- Github: [@luqmanmusah](https://github.com/luqmanmusah)
+- Email:[Musah Luqman](musahluqman@gmail.com)
+- Twitter: [@luqboi](https://twitter.com/luqboi)
+- LinkedIn:[@luqmanmusah](https://www.linkedin.com/luqmanmusah)
+
+## 🤝 Contributing
+
+Contributions, issues and feature requests are welcome!
+
+Feel free to check the [issues page](issues/).
```
Comment (arslanbisharat) [inline README.md]: - [ ] It would be great if you add the proper issue link

File: README.md [hunk 6]
```diff
@@ -1 +1,119 @@
...
+
+## 📝 License
+
+This project is [MIT](lic.url) licensed.
```
Comment (arslanbisharat) [inline README.md]: - [ ]  It would be great if you add the proper link here too.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested 💀 

Hello Developer, 👋 

Thank You for submitting this project.  I am very happy to see this project.  You have done an amazing job so far. 
-  The Pull Request is descriptive.  [Optional] You can add bullet points too. 
-  All tests working fine. This is a very rare case in which student submit the project with 100% tests passed



- Code is clean. 
- Linters working fine.

**Changes** 🔴 
- [ ] If you can make some changes requested below in the Readme file, your project is ready to be approved from a TSE. 

_When you make the changes please request another review. 👍 If you have another question, please reach out to me on slack @arslanbisharat._ 

**Best,
Arslan Bisharat** ✋

**Comment by mimipeshy** [comment]
### STATUS :APPROVED :partying_face: 

Hey there @luqmanmusah and @CollinsTatang  :wave: :wave:  good job on working on the requested changes. This project is therefore approved. Go ahead and merge. All the best in the next milestones :man_technologist: 

Regards
Peris Ndanu

**Pull request merged**